server/routes/downloadNomisConfig.ts

```diff
@@ -2,22 +2,59 @@ import type { RequestHandler, Response } from 'express'
 
 import type { ReferenceCode } from '../data/prisonApi'
 
-function streamReferenceCodeResponse(fileName: string, referenceCodes: ReferenceCode[], res: Response): void {
+type CsvCellValue = string | number | boolean | null
+
+function streamCsvDownload(res: Response, fileName: string, lines: Generator<CsvCellValue[]>): void {
```

Comment:
tbh, the generator could be an internal implementation of `streamCsvDownload()`, e.g.

```TS
// as before
streamCsvDownload(res, 'incident-types.csv', referenceCodes)
// [...]
for (const line of mapCodes(lines)) {
  // [...]
}
```

it could even be a nested function if we feel particularly courageous (still figuring out if they're a good idea or not 😝 )

Reply:
i thought it made the row mapping function _bodies_ easier to read with separate yields for each row. could just as easily build a giant array of all rows. it's what i did at first but this looks a bit better i think.